elevation_layer/src/elevation_layer.cpp

```diff
@@ -0,0 +1,242 @@
+//
+// Created by eugenio on 16.10.18.
+//
+
+#include <elevation_layer.h>
+#include <pluginlib/class_list_macros.h>
+
+PLUGINLIB_EXPORT_CLASS(elevation_layer::ElevationLayer, costmap_2d::Layer)
+
+using costmap_2d::NO_INFORMATION;
+using costmap_2d::LETHAL_OBSTACLE;
+using costmap_2d::FREE_SPACE;
+
+using costmap_2d::ObservationBuffer;
+using costmap_2d::Observation;
+
+namespace elevation_layer
+{
+
+    ElevationLayer::ElevationLayer() : filterChain_("grid_map::GridMap")
+    {
+        costmap_ = NULL;  // this is the unsigned char* member of parent class Costmap2D.
+    }
+
+    void ElevationLayer::onInitialize() {
+        ros::NodeHandle nh("~/" + name_), g_nh;
+        rolling_window_ = layered_costmap_->isRolling();
+
+        ElevationLayer::matchSize();
+        current_ = true;
```

Comment:
Value of `current_` is used by `move_base` to know if the layer is up-to-date ([look here](https://github.com/ANYbotics/navigation_ros/blob/33f95e00b81972ec37eeec7738bd21d8136a71b9/move_base/src/move_base.cpp#L858)). Therefore it should be updated depending on elevation layer updates. For example if we have not received elevation map for certain period we should make this value to false so  that `move_base` gives a warning and stops the robot.